src/components/Inputs.js

```diff
@@ -102,17 +154,24 @@ export function useTicketInput({ validators = [], ...rest }) {
         mono: true,
         ...rest,
       },
-    ])
+    ]),
+    ({ error, pass }) => ({
+      accessory: error ? (
+        <AccessoryIcon.Failure />
+      ) : deriving ? (
+        <AccessoryIcon.Pending />
+      ) : pass ? (
+        <AccessoryIcon.Success />
+      ) : null,
+    })
   );
-
-  return { ...input, accessory: input.pass ? <AccessoryIcon.Success /> : null };
 }
 
 const kPointValidators = [validatePoint, validateNotEmpty];
 const kPointTransformers = [prependSig];
 export function usePointInput(rest) {
   const [lastValidPoint, setLastValidPoint] = useState('');
-  const input = firstOf(
+  return useFirstOf(
     useForm([
```

Comment:
sure!

useForm returns a _list_ of inputs (which is required to handle dynamic forms like the InviteEmail one). In the case where we just want a single input we still use the logic from useForm but make the output much easier to consume for the components that just have the single input to care about.

truthfully we could probably convert all of the components with multiple inputs on them into a single `useForm()` state, but unless there are dynamic fields or more complex interactions between individual inputs the effort won't be worth it.

Reply:
firstOf was originally just a simple function that pulled the first input from the useForm (`useForm().inputs[0]`) but now that I've added the `setValue` callback I converted it to a hook so that we could use useCallback, etc.